rfcs/Subworlds.md

```diff
@@ -0,0 +1,139 @@
+        // provide a RunCriteria
+        .add_world_set(
+            WorldSet::new()
+                .with_run_criteria(some_criteria.system())
+                .with_system(another_system.system())
+                .with_worlds(["World2"])
+        )
+        .run();
+}
+```
+
+- By default, there is a single `World` contained within the `Worlds` (this mirrors the previous behavior of `bevy`).
+- Queries and systems still only operate entities and components from a single `World`.
+  - However, systems can be applied (concurrently) to multiple `Worlds` at a time.
+
+## Drawbacks 🙁
+
+In theory adding a dynamic number of `Subworlds` would add overhead since they must be kept in (something like) a `HashMap<Id, Subworld>`, where as having a single _global_ `World` does not incur this overhead.  
+
+## Rationale and alternatives 📜
+
+- Using _marker-components_ to distinguish entities that reside in disjoint sets. 
+  - This works, however, if `EntitiesA` {E1, E2, E3} are completely disjoint from `EntitiesB` {E4, E5, E6}, you shouldn't have to keep them grouped into a single set (`World`).
+  - Also, when constructing queries, this adds the overhead of need to check all entities instead of the entities of a specific `Subworld` that you are interested in. 
+
+- _Relations_ could also be a potential solution to this problem. 
+  - However the same problems described above apply. 
+  - The overhead of needing to check entity relationships exceeds that of running a query on a specific `Subworld`.  
+  - Overuse of _Relations_ and _marker-components_ could also lead to extreme archetypal fractaling, and worse case a single archetype-per-entities scenario. 
+
+## Prior art 🎨
+
+- See [here](https://discord.com/channels/691052431525675048/749335865876021248/834310334264115210) for the initial discord discussion that sparked this RFC.
+
+- The popular C++ ECS library [EnTT](https://github.com/skypjack/entt) supports this behavior, and even encourages it, via giving you direct access their `World` structure (`EnTT` calls these _registries_).
+```cpp
+auto level1 = entt::registry{}; // registry is analogous to bevy's World.
+auto level2 = entt::registry{};
+
+// depending on the current level, we can run the same system on a different sub-<worlds/registries>.
+some_system(level1);
+some_system(level2);
+```
+  - Currently supporting this is `bevy` is not possible, as systems and the `World` they operate on are closely coupled.
+  - `EnTT` is able to support this since systems are nothing more than free functions.
+
+## Unresolved questions ❓
+
+- How do resources behave in respect to each `Subworld`? Are they shared between them or made exclusive to each. (or both?? E.g. `Res<T>`/`ResMut<T>` vs `Local<T>`).
+- How do we support moving/copying entities and their component from one `Subworld` to another?
+- Should/Would it be possible for `Subworlds` to communicate? Or more specifically, the entities inside them to communicate?
```

Comment:
Perhaps via shared resources? This is particularly compelling since Events are stored as resources.